source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -840,9 +840,9 @@ void TunnelingConnectionPoolImpl::onPoolFailure(Http::ConnectionPool::PoolFailur
                                                 absl::string_view failure_reason,
                                                 Upstream::HostDescriptionConstSharedPtr host) {
   upstream_handle_ = nullptr;
-  callbacks_->onStreamFailure(reason, failure_reason, host);
   downstream_info_.upstreamInfo()->setUpstreamHost(host);
   downstream_info_.upstreamInfo()->setUpstreamTransportFailureReason(failure_reason);
+  callbacks_->onStreamFailure(reason, failure_reason, host);
```

Comment:
Does this change have a reason? If not can you revert it. If so please add a comment.

Reply:
Calling `callbacks_->onStreamFailure(reason, failure_reason, host);` could potentially remove the session, which will cause envoy to crash when we call `downstream_info_.upstreamInfo()->setUpstreamHost(host);` as downstream_info_ will be freed because of deleting the session.